File: monorepo_ruby/apps/web/cycle_b.rb
# C1.b — the `require_relative "cycle_a"` here would deadlock if loaded
# first, but Ruby's require caching makes the cycle safe as long as the
# CycleA constant is referenced LAZILY inside the method body.

class CycleB
  def initialize(tag); @tag = tag; end
  def bounce_to_alpha
    require_relative "cycle_a"     # lazy — avoids top-level cycle deadlock
    CycleA.new("bounce-from-#{@tag}").describe
  end
end

File: monorepo_go/apps/web/cycle_b.go
package main

// C1.b — cycle partner for cycle_a.go.

type Bravo struct {
	tag   string
	owner *Alpha                          // C1.b.type_ref → cycle_a.go
}

func newBravo(tag string) *Bravo        { return &Bravo{tag: tag} }

func (b *Bravo) bounceToAlpha() string {
	return newAlpha("bounce-from-" + b.tag).describe()
}
